### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by hilberturi**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -58,46 +58,88 @@ Purpose:  This project will show you the difference between member functions and
...
     {
+        // I will enforce safe pointer usage here as directed by 13)
+        if (a == nullptr || b == nullptr)
+        {
+            return nullptr;
+        }
         if( a->value < b->value ) return a;
         if( a->value > b->value ) return b;
+        // both values are equal
         return nullptr;
```
Comment (matkatmusic) [inline main.cpp]: D.R.Y.
you have 2 identical return statements.
- [x] revise the logic

File: main.cpp [hunk 2]
```diff
@@ -58,46 +58,88 @@ Purpose:  This project will show you the difference between member functions and
...
+        // plain member access without this->
+        // obviously a nullptr check for this is not necessary so I have deleted this
+        // check from the code that I copied from the static method.
         
+        // check for nullptr as guided by 13)
+        if (updatedValue == nullptr)
+        {
+            std::cout << "U::updateValuesAndMultiply(): cannot update for nullptr" << std::endl;
+            return adjustedValue * exactTargetValue;
+        }
```
Comment (matkatmusic) [inline main.cpp]: - [x] revise the logic: identical return statements used

File: main.cpp [hunk 3]
```diff
@@ -58,46 +58,88 @@ Purpose:  This project will show you the difference between member functions and
...
-             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
+             write something that makes the distance between that->adjustedValue and that->exactTargetValue get smaller
              */
-            that-><#name2#> += ;
+            that->adjustedValue += 0.5f * (that->exactTargetValue - that->adjustedValue);
         }
-        std::cout << "U's <#name2#> updated value: " << that-><#name2#> << std::endl;
-        return that-><#name2#> * that-><#name1#>;
+        std::cout << "U's adjustedValue updated value: " << that->adjustedValue << std::endl;
+        return that->adjustedValue * that->exactTargetValue;
```
Comment (matkatmusic) [inline main.cpp]: - [x] revise the logic: identical return statements used

File: main.cpp [hunk 4]
```diff
@@ -117,17 +159,25 @@ struct <#structname2#>
...
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    TPtrComparator f;                                            //7
+    auto* smaller = f.compare( &t1, &t2);                              //8
+    if (smaller != nullptr)
+    {
+        std::cout << "the smaller one is << " << smaller->name << std::endl; //9    
+    }
+    else 
+    {
+        std::cout << "'f.compare()' returned nullptr (one arg is nullptr, or both args have same value)"
```
Comment (matkatmusic) [inline main.cpp]: `one arg is nullptr` is not 100% correct.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
There are several duplicate return statements, which means the logic used in the code can be improved.

File: main.cpp (hunk 4 above)
Comment (hilberturi) [reply main.cpp]: yes both args can also be nullptrs

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**